lib/SyTest/Synapse.pm

```diff
@@ -187,32 +205,66 @@ sub start
       : "$self->{synapse_dir}"
    );
 
-   my @command = ( $self->{python} );
+   my @synapse_command = ( $self->{python} );
 
    if( $self->{coverage} ) {
       # Ensures that even --generate-config has coverage reports. This is intentional
-      push @command,
+      push @synapse_command,
          "-m", "coverage", "run", "-p", "--source=$self->{synapse_dir}/synapse";
    }
 
-   push @command,
+   push @synapse_command,
       "-m", "synapse.app.homeserver",
       "--config-path" => $config_path,
       "--server-name" => "localhost:$port";
 
    $output->diag( "Generating config for port $port" );
 
+   my @config_command = (
+      @synapse_command, "--generate-config", "--report-stats=no"
+   );
+
+   my @command;
+
+   if( $self->{dendron} ) {
+      $db_type eq "pg" or die "Dendron can only run against postgres";
+
+      my @db_arg_pairs = ();
+      while( my ( $key, $value ) = each %db_args ) {
```

Comment:
Don't use `each()` on a hash. It's silly and needs to be taken out and shot.

Instead, especially in this case, you might find `List::Util::pairmap` useful. It's like a regular `map` only takes its input list as an even-sized list of pairs, assigning values to `$a` and `$b` for each iteration:

    use List::Util qw( pairmap );

    my @db_arg_pairs = pairmap { $a eq "database" ? "dbname=$b" : "$a=$b" } %db_args;